qutip/qobj.py

```diff
@@ -2216,10 +2219,10 @@ def _ptrace_dense(Q, sel):
                           .reshape(rd + rd)
                           .transpose(qtrace + [nd + q for q in qtrace] +
                                      sel + [nd + q for q in sel])
-                          .reshape([np.prod(dtrace),
-                                    np.prod(dtrace),
-                                    np.prod(dkeep),
-                                    np.prod(dkeep)]))
+                          .reshape([np.prod(dtrace, dtype=np.int32),
```

Comment:
Thank you for the quite patch.
Could you please add the `dtype` to the ket case also (line 2215).

Reply:
Oh yeah, good point.  Fixed that, and force-updated the PR.